ui/src/main/main.module.scss

```diff
@@ -33,3 +33,21 @@
         z-index: 3;
     }
 }
+
+.main-overlay {
+    visibility: hidden;
+    opacity: 0;
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    background-color: vayla-design.$color-white-default;
+    z-index: 9999;
+    transition: opacity 0.5s;
+}
+
+.main-overlay-enabled {
+    visibility: visible;
+    opacity: 0.75;
+}
```

Comment:
Overlay on html-elementtinä olemassa aina, jotta fade-in toimii nätisti. Visibility: hidden tekee taas sen efektin, ettei tämä divi kuitenkaan blokkaa esim. klikkauksia jne.